File: server/src/main/java/org/dan/jadalnia/app/user/UserRegRequest.java
package org.dan.jadalnia.app.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.dan.jadalnia.app.festival.pojo.Fid;

import javax.validation.constraints.Size;
import java.util.Optional;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserRegRequest {
    Fid festivalId;
    @Size(min = 3, max = 40)
    String name;
    String session;
    private UserType userType;

    Optional<String> email;
    Optional<String> phone;

    public static class UserRegRequestBuilder {
        Optional<String> email = Optional.empty();
        Optional<String> phone = Optional.empty();
    }
}

File: server/src/main/kotlin/org/dan/jadalnia/app/user/UserRegRequest.kt
package org.dan.jadalnia.app.user

import com.fasterxml.jackson.annotation.JsonCreator
import org.dan.jadalnia.app.festival.pojo.Fid

import javax.validation.constraints.Size
import java.util.Optional


data class UserRegRequest @JsonCreator constructor(
        var festivalId: Fid,
        @Size(min = 3, max = 40)
        var name: String,
        var session: String,
        var userType: UserType) //,
// todo how to have default values
//        var email: Optional<String> = Optional.empty(),
//        var phone: Optional<String> = Optional.empty())
{
//    @JsonCreator
//    constructor(
//            festivalId: Fid,
//            name: String,
//            session: String,
//            userType: UserType): this(festivalId, name, session, userType,
//            Optional.empty(), Optional.empty()) {
//    }
}
